hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieIndex.java

```diff
@@ -111,23 +104,17 @@ private static Stream<Arguments> indexTypeParams() {
         {IndexType.GLOBAL_BLOOM, true, false},
         {IndexType.SIMPLE, true, true},
         {IndexType.SIMPLE, true, false},
-        {IndexType.SIMPLE, false, true},
-        {IndexType.SIMPLE, false, false},
         {IndexType.GLOBAL_SIMPLE, true, true},
         {IndexType.GLOBAL_SIMPLE, true, false},
-        {IndexType.GLOBAL_SIMPLE, false, true},
-        {IndexType.GLOBAL_SIMPLE, false, false},
-        {IndexType.BUCKET, false, true},
+        {IndexType.BUCKET, true, false},
         {IndexType.BUCKET, false, false},
-        {IndexType.RECORD_INDEX, true, true},
-        {IndexType.RECORD_INDEX, true, false}
+        {IndexType.RECORD_INDEX, true, true}
```

Comment:
- popluateMetaFields (true/false) only applies to bucket index.
- only test record index when MDT enabled